estate/models/estate_property.py

```diff
@@ -150,3 +150,9 @@ def action_cancel(self):
             "Selling price must be strictly positive",
         )
     ]
+
+    @api.ondelete(at_uninstall=False)
+    def unlink_if_not_new_or_cancelled(self):
+        for record in self:
+            if record.state not in ('new', 'canceled'):
```

Comment:
Although this is technically correct, we can use the operator "in" to iterate over a tuple 

You will often see domains like this: `domain = [('name','in', ['alexis', 'cindy', ...])]`

I would suggest that you try to use a list instead, so you don't get confused later

But i'll leave up to you

```suggestion
            if record.state not in ['new', 'canceled']:
```